src/portal_visualization/builders/anndata_builders.py

```diff
@@ -71,6 +73,48 @@ def get_conf_cells(self, marker=None):
         cell_set_obs.append("leiden")
         cell_set_obs_names.append("Leiden")
         gene_alias = 'var/hugo_symbol' if 'var' in z and 'hugo_symbol' in z['var'] else None
+        if (gene_alias is not None and marker is not None):
+            # If user has indicated a marker gene in parameters and we have a hugo_symbol mapping,
+            # then we need to convert it to the proper underlying ensembl ID for the dataset
+            # in order for the views to reflect the correct gene.
+
+            obs_attrs = z["obs"].attrs.asdict()
+            encoding_version = obs_attrs["encoding-version"]
+
+            # Encoding Version 0.1.0
+            # https://anndata.readthedocs.io/en/0.7.8/fileformat-prose.html#categorical-arrays
+            if (encoding_version == "0.1.0"):
+                # Get the list of ensembl IDs from the zarr store
+                ensembl_ids_key = z['var'].attrs['_index']
+                ensembl_ids = z['var'][ensembl_ids_key]
+                # Get the list of hugo symbols
+                hugo_symbols = z[gene_alias]
+                # Indices in hugo_index_list match to indices of mapped ensembl ID's
+                # Values in hugo_index_list match to indices in hugo_categories
+                hugo_index_list = hugo_symbols[:]
+                # Get the key for the hugo categories list from the categorical entry attributes
+                hugo_categories_key = hugo_symbols.attrs.asdict()['categories']
+                # Get the list of categories that the hugo_index_list's values map to
+                hugo_categories = z['var'][hugo_categories_key][:]
+                # Find the index of the user-provided marker gene in the list of hugo symbols
+                marker_index_in_categories = np.where(hugo_categories == marker)[0][0]
```

Comment:
```suggestion
                is_marker_index_in_categories = np.where(hugo_categories == marker)[0][0]
```
slight preference for naming booleans with verbs

Reply:
I agree with the use of `is` for booleans, but this variable holds the actual found index value, so I've reverted this change. Finding this index is one of the steps of the reverse lookup in this categorical encoding: https://anndata.readthedocs.io/en/0.7.8/fileformat-prose.html#dataframes

- `marker_index_in_categories` finds the index of the HUGO value in the categories list
- `marker_index` finds the `marker_index_in_categories` in the list that maps ENSEMBL ID's to HUGO values' indices in the categories list
- if `marker_index` exists (i.e. if there is an EMSEMBL ID mapped to this HUGO symbol) then marker is updated to the corresponding ENSEMBL ID

I tried to keep this clear via the inline comments, but can expand on them if needed.

Reply:
Ok, that's fine.  My mistake.